lib/index.js

```diff
@@ -0,0 +1,241 @@
+const validate = require("./validate");
+const wasm = require("./secp256k1");
+
+const WASM_BUFFER = new Uint8Array(wasm.memory.buffer);
+const WASM_PRIVATE_KEY_PTR = wasm.PRIVATE_INPUT.value;
+const WASM_PUBLIC_KEY_INPUT_PTR = wasm.PUBLIC_KEY_INPUT.value;
+const WASM_PUBLIC_KEY_INPUT_PTR2 = wasm.PUBLIC_KEY_INPUT2.value;
+const WASM_TWEAK_INPUT_PTR = wasm.TWEAK_INPUT.value;
+const WASM_HASH_INPUT_PTR = wasm.HASH_INPUT.value;
+const WASM_EXTRA_DATA_INPUT_PTR = wasm.EXTRA_DATA_INPUT.value;
+const WASM_SIGNATURE_INPUT_PTR = wasm.SIGNATURE_INPUT.value;
+
+const PRIVATE_KEY_INPUT = WASM_BUFFER.subarray(
+  WASM_PRIVATE_KEY_PTR,
+  WASM_PRIVATE_KEY_PTR + validate.PRIVATE_KEY_SIZE
+);
+const PUBLIC_KEY_INPUT = WASM_BUFFER.subarray(
+  WASM_PUBLIC_KEY_INPUT_PTR,
+  WASM_PUBLIC_KEY_INPUT_PTR + validate.PUBLIC_KEY_UNCOMPRESSED_SIZE
+);
+const PUBLIC_KEY_INPUT2 = WASM_BUFFER.subarray(
+  WASM_PUBLIC_KEY_INPUT_PTR2,
+  WASM_PUBLIC_KEY_INPUT_PTR2 + validate.PUBLIC_KEY_UNCOMPRESSED_SIZE
+);
+const TWEAK_INPUT = WASM_BUFFER.subarray(
+  WASM_TWEAK_INPUT_PTR,
+  WASM_TWEAK_INPUT_PTR + validate.TWEAK_SIZE
+);
+const HASH_INPUT = WASM_BUFFER.subarray(
+  WASM_HASH_INPUT_PTR,
+  WASM_HASH_INPUT_PTR + validate.HASH_SIZE
+);
+const EXTRA_DATA_INPUT = WASM_BUFFER.subarray(
+  WASM_EXTRA_DATA_INPUT_PTR,
+  WASM_EXTRA_DATA_INPUT_PTR + validate.EXTRA_DATA_SIZE
+);
+const SIGNATURE_INPUT = WASM_BUFFER.subarray(
+  WASM_SIGNATURE_INPUT_PTR,
+  WASM_SIGNATURE_INPUT_PTR + validate.SIGNATURE_SIZE
+);
+
+function assumeCompression(compressed, p) {
+  return compressed === undefined
+    ? p.length
+    : compressed === true
+    ? validate.PUBLIC_KEY_COMPRESSED_SIZE
+    : validate.PUBLIC_KEY_UNCOMPRESSED_SIZE;
+}
+
+function isPoint(p) {
+  if (!validate.isPoint(p)) return false;
+
+  try {
+    PUBLIC_KEY_INPUT.set(p);
+    return wasm.isPoint(p.length) === 1;
+  } finally {
+    PUBLIC_KEY_INPUT.fill(0);
+  }
+}
+
+function isPointCompressed(p) {
+  return isPoint(p) && p.length === validate.PUBLIC_KEY_COMPRESSED_SIZE;
+}
+
+function isPrivate(x) {
+  return validate.isPrivate(x);
+}
+
+function pointAdd(pA, pB, compressed) {
```

Comment:
Will pA be modified in place?

Reply:
No, we can not modify JS values from WASM while they do not refer to WASM memory. `pA` copied to input WASM buffer, and if addition successful we return new `Uint8Array`.